presto-main/src/main/java/com/facebook/presto/sql/gen/InCodeGenerator.java

```diff
@@ -61,6 +70,28 @@ public ByteCodeNode generateExpression(Signature signature, ByteCodeGeneratorCon
         Type type = arguments.get(0).getType();
         Class<?> javaType = type.getJavaType();
 
+        SwitchGenerationCase switchGenerationCase = SwitchGenerationCase.LARGE_LIST;
+        if (values.size() <= 1000) {
+            boolean areSmallPrimitives = false;
+            if (javaType == long.class) {
```

Comment:
Use `type instanceof BigintType || type instanceof DateType` instead. Some types have special semantics for equals. Therefore, it's not safe to do this optimization in all cases where `javaType == long.class`.